payments.md

```diff
@@ -110,16 +47,173 @@ type SpendVoucher struct {
     // Lane specifies which 'lane' of the payment channel this voucher is for.
     // Lanes may be either open or closed, a voucher for a closed lane may not be redeemed
     Lane uint64
-    
-    // Merges specifies a list of lane-nonce pairs that this voucher will close. 
+
+    // Merges specifies a list of lane-nonce pairs that this voucher will close.
     // This voucher may not be redeemed if any of the lanes specified here are already
-    // closed, or their nonce specified here is lower than the nonce on-chain.
-    Merges []Pair<uint64, uint64>
+    // closed, or their nonce specified here is lower than the nonce of the lane on-chain.
+    Merges []MergePair
+
+    TimeLock uint64
+
+    SecretPreimage []byte
+
+    RequiredSector []byte
+  
+    DataCommitment []byte
+
+    MinCloseHeight uint64
     
     Sig Signature
 }
+
+type MergePair struct {
+    Lane uint64
+    Nonce uint64
+}
 ```
 
+```go
+type PaymentChannel struct {
+    From Address
+    To Address
+
+    ChannelTotal TokenAmount
+    ToSend TokenAmount
+
+    ClosingAt uint64
+    MinCloseHeight uint64
+
+    LaneStates map[uint64]LaneState
+}
+
+type LaneState struct {
+    Nonce uint64
+    Redeemed TokenAmount
+}
+
+func (paych *PaymentChannel) validateSignature(sv SpendVoucher) {
+    if msg.From == paych.From {
+        ValidateSignature(sv.SerializeNoSig(), sv.Signature, paych.To)
+    } else if msg.From == paych.To {
+        ValidateSignature(sv.SerializeNoSig(), sv.Signature, paych.From)
+    } else {
+        Fatal("bad programmer")
+    }
+}
+
+func (paych *PaymentChannel) UpdateChannelState(sv SpendVoucher, secret []byte, pip *PieceInclusionProof) {
+    if !paych.validateSignature(sv) {
+        Fatal("Signature Invalid")
+    }
+
+    if chain.Now() < sv.TimeLock {
+        Fatal("cannot use this voucher yet!")
+    }
+
+    if sv.SecretPreimage != nil {
+        if Hash(secret) != sv.SecretPreimage {
+            Fatal("Incorrect secret!")
+        }
+    }
+  
+  	if sv.DataCommitment != nil {
```

Comment:
see my previous comment, my opinion is that either this contract is renamed or it should be more generic.